protocols/gossipsub/src/time_cache.rs

```diff
@@ -36,6 +36,10 @@ struct ExpiringElement<Element> {
     expires: Instant,
 }
 
+#[deprecated(
+    since = "0.45.0",
+    note = "`time_cache` module and all it's included public structs are going to be deprecated in future release."
```

Comment:
This isn't _quite_ true. The message should say that this is an internal type and is not meant to be used by users.